packages/common/src/llm/openai.ts

```diff
@@ -48,19 +44,32 @@ export async function generateContent<M extends string>(
     })
   }
 
-  const response = await openAIClient.chat.completions.create({
-    model: input.model,
-    max_tokens: input.maxTokens || undefined, // note: ignore a zero value as the Studio doesn't support empty number inputs and defaults this to 0
-    temperature: input.temperature,
-    top_p: input.topP,
-    response_format: input.responseFormat === 'json_object' ? { type: 'json_object' } : undefined,
-    // TODO: the Studio is adding an empty item by default in the action input form
-    stop: input.stopSequences?.filter((x) => x.trim()), // don't send empty values
-    user: input.userId || undefined, // don't send an empty value
-    messages,
-    tool_choice: mapToOpenAIToolChoice(input.toolChoice), // note: the action input type is
-    tools: mapToOpenAITools(input.tools),
-  })
+  let response: OpenAI.Chat.Completions.ChatCompletion
+
+  try {
+    response = await openAIClient.chat.completions.create({
+      model: input.model,
+      max_tokens: input.maxTokens || undefined, // note: ignore a zero value as the Studio doesn't support empty number inputs and defaults this to 0
+      temperature: input.temperature,
+      top_p: input.topP,
+      response_format: input.responseFormat === 'json_object' ? { type: 'json_object' } : undefined,
+      // TODO: the Studio is adding an empty item by default in the action input form
+      stop: input.stopSequences?.filter((x) => x.trim()), // don't send empty values
+      user: input.userId || undefined, // don't send an empty value
+      messages,
+      tool_choice: mapToOpenAIToolChoice(input.toolChoice), // note: the action input type is
+      tools: mapToOpenAITools(input.tools),
+    })
+  } catch (err) {
+    logger
+      .forBot()
+      .error(
+        `generateContent action failed because the "${
+          params.provider
+        }" LLM provider returned an error: ${JSON.stringify(err)}`
```

Comment:
From my experience with integrations, I found helpful to log the complete error object (apart from a small "presenting" message before the error to make it clear to the user what it is that we're logging) because sometimes the other end can return an ambiguous or cryptic error message but clarifying details (like HTTP/error code) are present in the complete error object.

Reply:
Addressed in https://github.com/botpress/botpress/pull/13159/commits/cffe61aa70cae276cecb09d8a3d7dfb8a4c90499